### PR TITLE
DOMA-1509 Resident ticket review

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**
[Jira task](https://doma.atlassian.net/browse/DOMA-1509)

1) Added `reviewValue` and `reviewComment` fields to `Ticket` schema
2) Updated `TicketChanges`
3) Added `statusReopenedCounter` increment logic when ticket moves from "complete" to "in progress" status
4) Added reopened ticket bage to `/ticket` and `/ticket/[id]` pages
5) Added `closeCompletedTicketsTask` cron task, which сloses tickets that are in the "completed" status for 7 days
6) Added `reviewValue`, `reviewComment` and `statusReopenedCounter` to excel and filters

### CODE REVIEW

File: apps/condo/domains/meter/schema/MeterReading.js [hunk 1]
```diff
@@ -77,7 +77,6 @@ const MeterReading = new GQLListSchema('MeterReading', {
     },
     hooks: {
         resolveInput: async ({ operation, listKey, context, resolvedData, existingItem }) => {
-            await triggersManager.executeTrigger({ operation, data: { resolvedData, existingItem }, listKey }, context)
```
Comment (pahaz) [inline apps/condo/domains/meter/schema/MeterReading.js:80]: Ok, no triggers for meter readings :)

File: apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx [hunk 2]
```diff
@@ -12,6 +12,9 @@ import { MAX_DESCRIPTION_DISPLAY_LENGTH } from '@condo/domains/ticket/constants/
 import { FormattedMessage } from 'react-intl'
 import { fontSizes } from '@condo/domains/common/constants/style'
 import dayjs from 'dayjs'
+import isNil from 'lodash/isNil'
+import { getReviewMessageByValue } from '../../utils/clientSchema/Ticket'
+const { RESIDENT } = require('@condo/domains/user/constants/common')
```
Comment (pahaz) [inline apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx:17]: import / require ?

File: apps/condo/domains/ticket/schema/Ticket.js [hunk 3]
```diff
@@ -308,21 +320,36 @@ const Ticket = new GQLListSchema('Ticket', {
...
+                    const existedStatus = await TicketStatus.getOne(context, { id: existedStatusId })
+                    const resolvedStatus = await TicketStatus.getOne(context, { id: resolvedStatusId })
+
+                    if (get(existedStatus, 'type') === COMPLETED_STATUS_TYPE && get(resolvedStatus, 'type') === NEW_OR_REOPENED_STATUS_TYPE) {
+                        const existedStatusReopenedCounter = existingItem['statusReopenedCounter']
+                        resolvedData['statusReopenedCounter'] = existedStatusReopenedCounter + 1
+                        resolvedData['executor'] = null
+                    }
+                }
             }
```
Comment (pahaz) [inline apps/condo/domains/ticket/schema/Ticket.js:347]: it's better to extract this logic to function

File: apps/condo/domains/ticket/schema/Ticket.js [hunk 4]
```diff
@@ -58,6 +60,16 @@ const Ticket = new GQLListSchema('Ticket', {
                 create: false,
             },
         },
+        reviewValue: {
+            schemaDoc: 'Review of the ticket by a resident on a 2-point scale. 0 – ticket returned, 1 – bad review, 2 – good review',
+            type: Select,
+            dataType: 'string',
+            options: '0, 1, 2',
```
Comment (pahaz) [inline apps/condo/domains/ticket/schema/Ticket.js:67]: is it a string? probably it's better to use some keyword. good, bad, ..

**Review by sitozzz — APPROVED** [review]

File: apps/condo/domains/ticket/schema/Ticket.js (hunk 4 above)
Comment (pahaz) [reply apps/condo/domains/ticket/schema/Ticket.js:67]: or use the numbers

File: apps/condo/domains/ticket/hooks/useTicketTableFilters.tsx [hunk 5]
```diff
@@ -91,6 +96,11 @@ export function useTicketTableFilters (): Array<FiltersMeta<MeterReadingWhereInp
         { label: PaidMessage, value: 'isPaid' },
         { label: EmergencyMessage, value: 'isEmergency' },
         { label: WarrantyMessage, value: 'isWarranty' },
+        { label: ReturnedMessage.toLowerCase(), value: 'statusReopenedCounter' },
+    ]
+    const reviewValueOptions = [
+        { label: GoodReviewMessage, value: '2' },
```
Comment (sitozzz) [inline apps/condo/domains/ticket/hooks/useTicketTableFilters.tsx:102]: May value of review be stored as a more readable datatype (like `enum`)

File: apps/condo/domains/ticket/schema/Ticket.js (hunk 4 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/ticket/schema/Ticket.js]: used enums 'returned, good, bad'

File: apps/condo/domains/ticket/hooks/useTicketTableFilters.tsx (hunk 5 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/ticket/hooks/useTicketTableFilters.tsx]: used enums 'returned, good, bad'

File: apps/condo/domains/ticket/schema/Ticket.js [hunk 6]
```diff
@@ -59,6 +58,15 @@ const Ticket = new GQLListSchema('Ticket', {
                 create: false,
             },
         },
+        reviewValue: {
+            schemaDoc: 'Review of the ticket by a resident on a 2-point scale. 0 – ticket returned, 1 – bad review, 2 – good review',
+            type: Select,
+            options: 'returned, bad, good',
```
Comment (pahaz) [inline apps/condo/domains/ticket/schema/Ticket.js:64]: it's better to extract it to const like in other cases: User.type

File: apps/condo/domains/ticket/utils/serverSchema/resolveHelpers.js [hunk 7]
```diff
@@ -2,7 +2,33 @@ const get = require('lodash/get')
...
+            const resolvedStatus = await TicketStatus.getOne(context, { id: resolvedStatusId })
+
+            if (get(existedStatus, 'type') === COMPLETED_STATUS_TYPE && get(resolvedStatus, 'type') === NEW_OR_REOPENED_STATUS_TYPE) {
+                const existedStatusReopenedCounter = existingItem['statusReopenedCounter']
+                resolvedData['statusReopenedCounter'] = existedStatusReopenedCounter + 1
+                resolvedData['executor'] = null
+            }
+        }
+    }
+}
```
Comment (pahaz) [inline apps/condo/domains/ticket/utils/serverSchema/resolveHelpers.js:31]: it's too complex. You have here at least three different things:
 1) update ticket order
 2) calculate reopened counter
 3) clean deadline and clean executor (is it really needed?)

It's better to split it to three different functions

File: apps/condo/domains/ticket/utils/serverSchema/resolveHelpers.js [hunk 8]
```diff
@@ -2,7 +2,33 @@ const get = require('lodash/get')
...
+
+async function updateResolvedDataByResolvedStatus (context, existingItem, resolvedData) {
+    const existedStatusId = get(existingItem, 'status')
+    const resolvedStatusId = get(resolvedData, 'status')
+
+    if (resolvedStatusId) {
+        addOrderToTicket(resolvedData, resolvedStatusId)
+
+        if (TERMINAL_TICKET_STATUS_IDS.includes(resolvedStatusId)) {
+            resolvedData.deadline = null
```
Comment (pahaz) [inline apps/condo/domains/ticket/utils/serverSchema/resolveHelpers.js:17]: it's better not to clean deadline data.

Comment (nomerdvadcatpyat) [reply apps/condo/domains/ticket/utils/serverSchema/resolveHelpers.js:17]: created task for it https://doma.atlassian.net/browse/DOMA-2531

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/ticket/utils/serverSchema/resolveHelpers.js [hunk 9]
```diff
@@ -2,39 +2,31 @@ const get = require('lodash/get')
 const { getSectionAndFloorByUnitName } = require('@condo/domains/ticket/utils/unit')
 const { Property } = require('@condo/domains/property/utils/serverSchema')
 const { Contact } = require('@condo/domains/contact/utils/serverSchema')
-const { TICKET_ORDER_BY_STATUS, STATUS_IDS, TERMINAL_TICKET_STATUS_IDS } = require('@condo/domains/ticket/constants/statusTransitions')
+const { TICKET_ORDER_BY_STATUS, STATUS_IDS } = require('@condo/domains/ticket/constants/statusTransitions')
 const { COMPLETED_STATUS_TYPE, NEW_OR_REOPENED_STATUS_TYPE } = require('@condo/domains/ticket/constants')
 const { TicketStatus } = require('@condo/domains/ticket/utils/serverSchema')
 
-async function updateResolvedDataByResolvedStatus (context, existingItem, resolvedData) {
+function calculateTicketOrder (resolvedData, statusId) {
```
Comment (pahaz) [inline apps/condo/domains/ticket/utils/serverSchema/resolveHelpers.js:9]: it's better to mark it as `async`

**Pull request merged**